nautobot/extras/models/customfields.py

```diff
@@ -521,6 +521,26 @@ class Meta:
     def __str__(self):
         return self.label
 
+    @property
+    def choices_cache_key(self):
+        return f"nautobot.extras.customfield.choices.{self.pk}"
+
+    @property
+    def choices(self) -> list[str]:
+        """
+        Cacheable shorthand for retrieving custom_field_choices values associated with this model.
+
+        Returns:
+            list[str]: List of choice values, ordered by weight.
+        """
+        if self.type not in [CustomFieldTypeChoices.TYPE_SELECT, CustomFieldTypeChoices.TYPE_MULTISELECT]:
+            return []
+        choices = cache.get(self.choices_cache_key)
+        if choices is None:
+            choices = list(self.custom_field_choices.order_by("weight", "value").values_list("value", flat=True))
```

Comment:
Is CustomFieldChoice already ordered by `weight` and `value`? So perhaps we do not need to specify the ordering here?

Reply:
It's ordered by `custom_field, weight, value` by default. I *think* narrowing it down here is a bit more efficient even if it's not strictly necessary?